dora/src/bytecode/writer.rs

```diff
@@ -973,12 +981,24 @@ impl BytecodeWriter {
         self.emit_values(&values);
     }
 
+    pub fn set_position(&mut self, pos: Position) {
+        self.position = pos;
+    }
+
     fn emit_op(&mut self, inst: BytecodeOpcode) {
         let values = [inst as u32];
         self.emit_values(&values);
     }
 
+    fn emit_position(&mut self) {
+        let offset = self.code.len() as u32;
+        let position = (offset, self.position);
+        self.positions.push(position);
+    }
+
     fn emit_values(&mut self, values: &[u32]) {
+        self.emit_position();
```

Comment:
Right now we emit a new position for each and every bytecode. I would only emit a bytecode if 1) the bytecode needs a position and 2) the position is different than the last recorded one.

Reply:
For 1) we could have a method `BytecodeOpcode::needs_position()` that returns true for opcodes that need a position. We might want to change `emit_values` such that the opcode is not part of the array but a separate argument.

Reply:
ad 1) Sounds good. 
ad 2) Do you mean, because we could interpret the positions as range? Wouldn't that conflict with 1) as we do not set the position for every bytecode?

Reply:
Yes, they would be interpreted as ranges. How would this conflict? Assuming set_position is invoked - this would always be correct just use less memory.

Reply:
Conflict is maybe too strong. But personally I think one of those approaches would be sufficient enough. Additionally, I'm not sure, that ranges will occur very often.